pkg/actions/lua/lakefs/catalogexport/delta_exporter.lua

```diff
@@ -128,6 +134,14 @@ local function export_delta_log(action, table_def_names, write_object, delta_cli
                         elseif entry.remove ~= nil then
                             entry.remove.path = physical_path
                         end
+                    elseif code == 404 then
+                        if entry.remove ~= nil then
+                            -- If the object is not found, and the entry is a remove entry, we can assume it was vacuumed
+                            print(string.format("Object with path '%s' of a `remove` entry wasn't found. Assuming vacuum.", unescaped_path))
+                            unfound_paths[unescaped_path] = nil
```

Comment:
Isn't this redundant?

Reply:
If it was an `add` entry (meaning the clause beneath) and a stat object failed, then it means that there might be a `remove` entry with the same path further ahead which will cause a vacuum to delete it, and in that case we don't want to fail the process, thus we add it to the `unfound_paths` table. 
This line indicates that if a remove occurred and a stat object failed, we should treat any previously added entries as vacuumed objects.
The test on line 155 checks if there were any failed `add` entry path stat object requests (meaning that they weren't removed by a `remove` entry), and if that's the case throw an error.